posthog/session_recordings/synthetic_playlists.py

```diff
@@ -201,6 +205,44 @@ def to_synthetic_playlist(self) -> "SyntheticPlaylistDefinition":
         )
 
 
+@dataclass
+class ExpiringPlaylistSource(SyntheticPlaylistSource):
+    def get_session_ids(self, team: Team, user: User, limit: int | None = None) -> list[str]:
+        query = RecordingsQuery(limit=1000, order=RecordingOrder.RECORDING_TTL)
+        recordings, _, _ = list_recordings_from_query(query, user, team)
+
+        now = datetime.now(UTC)
+        ten_days_from_now = now + timedelta(days=10)
+
+        result = [r.session_id for r in recordings if r.expiry_time and now <= r.expiry_time <= ten_days_from_now]
+
+        if limit:
+            result = result[:limit]
+
+        return result
+
+    def count_session_ids(self, team: Team, user: User) -> int:
+        query = RecordingsQuery(limit=10000, order=RecordingOrder.RECORDING_TTL)
+        recordings, _, _ = list_recordings_from_query(query, user, team)
+
+        now = datetime.now(UTC)
+        ten_days_from_now = now + timedelta(days=10)
+
+        return sum(1 for r in recordings if r.expiry_time and now <= r.expiry_time <= ten_days_from_now)
```

Comment:
in  a perfect world we'd be able to filter by expiry to avoid this in-memory loop

but that can come tomorrow and tomorrow and tomorrow

(i want to address counting separately)